subcommand/acl-controller/command_test.go

```diff
@@ -421,3 +441,124 @@ func makeAuthMethod(principals interface{}) *api.ACLAuthMethod {
+			agentConfig: AgentConfig{Config: Config{Datacenter: "dc1"}},
+			expErr:      "both Config.PrimaryDatacenter and DebugConfig.PrimaryDatacenter are empty",
+		},
+		"mgw WAN fed not enabled": {
+			agentConfig: AgentConfig{Config: Config{Datacenter: "dc1", PrimaryDatacenter: "dc1"}},
+		},
+		"mgw WAN fed enabled, not primary DC": {
+			agentConfig: AgentConfig{Config: Config{Datacenter: "dc2"}, DebugConfig: Config{PrimaryDatacenter: "dc1"}},
+		},
+		"mgw WAN fed enabled, primary DC, policy exists": {
+			agentConfig: AgentConfig{
+				Config: Config{Datacenter: "dc1"},
+				DebugConfig: Config{
+					PrimaryDatacenter:               "dc1",
+					MeshGatewayWANFederationEnabled: true,
+				},
+			},
+			existingPolicy: true,
+			expPolicy:      expOSSAnonTokenPolicy,
+		},
+		"mgw WAN fed enabled, primary DC, create policy OSS": {
+			agentConfig: AgentConfig{
+				Config: Config{Datacenter: "dc1"},
+				DebugConfig: Config{
+					PrimaryDatacenter:               "dc1",
+					MeshGatewayWANFederationEnabled: true,
+				},
+			},
+			expPolicy: expOSSAnonTokenPolicy,
+		},
+		"mgw WAN fed enabled, primary DC, create policy ENT": {
+			agentConfig: AgentConfig{
+				Config: Config{Datacenter: "dc1"},
+				DebugConfig: Config{
+					PrimaryDatacenter:               "dc1",
+					MeshGatewayWANFederationEnabled: true,
+				},
+			},
+			partitionsEnabled: true,
+			expPolicy:         expEntAnonTokenPolicy,
+		},
+	}
+	for name, c := range cases {
+		c := c
+		t.Run(name, func(t *testing.T) {
+			cfg := testutil.ConsulServer(t, testutil.ConsulACLConfigFn)
+			consulClient, err := api.NewClient(cfg)
+			require.NoError(t, err)
+
+			cmd := Command{
+				log:                   hclog.Default().Named("acl-controller"),
+				flagPartitionsEnabled: c.partitionsEnabled,
+			}
+
+			// if we're simulating that the policy already exists, then create it.
+			var expPolicyID string
+			if c.existingPolicy {
+				policy, _, err := consulClient.ACL().PolicyCreate(&api.ACLPolicy{
+					Name:        anonPolicyName,
+					Description: "Anonymous token policy",
+					Rules:       c.expPolicy,
+				}, &api.WriteOptions{})
+				require.NoError(t, err)
+				expPolicyID = policy.ID
+			}
+
+			expAnonToken, _, err := consulClient.ACL().TokenRead(anonTokenID, &api.QueryOptions{})
+			require.NoError(t, err)
+
+			err = cmd.upsertAnonymousTokenPolicy(consulClient, c.agentConfig)
+
+			if len(c.expErr) == 0 {
+				require.NoError(t, err)
+				obsAnonToken, _, err := consulClient.ACL().TokenRead(anonTokenID, &api.QueryOptions{})
+				require.NoError(t, err)
+
+				if len(c.expPolicy) > 0 {
+					// if we expect the policy to be created then read it back to make sure
+					// it was and that it matches the expected
+					obsAnonTokenPolicy, _, err := consulClient.ACL().PolicyReadByName(anonPolicyName, &api.QueryOptions{})
+					require.NoError(t, err)
+					require.Equal(t, c.expPolicy, obsAnonTokenPolicy.Rules)
+
+					if expPolicyID == "" {
```

Comment:
nit: To simplify this slightly, we could ignore checking the specific policy id, which would allow us to remove this `expPolicyId` variable. It should be okay to just check the policy name, since Consul doesn't let you create two policies with the same name (in the same partition/namespace).

Not a blocker.

Reply:
OK, this suggestion is fine with me. This is how I originally had it but thought that checking the ID would be more thorough. But admittedly this check is kind of janky so I'm fine to remove it.